CMakeLists.txt

```diff
@@ -3627,6 +3666,78 @@ if(gRPC_INSTALL)
   )
 endif()
 
+endif()
+if(gRPC_BUILD_TESTS)
+
+add_library(re2
```

Comment:
re2 has its own cmake that we should use. https://github.com/google/re2/blob/master/CMakeLists.txt
So seeing the re2 sources listed in our CMakeLists.txt  seems suspicious.